### PR TITLE
guides: add ABI stability guide

### PR DESCRIPTION
**Pull request by gabrielschulhof**

### CODE REVIEW

**Comment by gabrielschulhof** [comment]
@mhdawson @gireeshpunathil @benjamingr I have moved the guide to this repo so it might appear on the Web site. Could you PTAL?

**Review by Trott — APPROVED** [review]

**Review by benjamingr — APPROVED** [review]
Hey, I didn't get a chance to respond to the feedback initially but this is _much_ better and I still really like having this guide.

Nice work! <3

**Review by gireeshpunathil — APPROVED** [review]

**Review by vsemozhetbyt — APPROVED** [review]

**Comment by gabrielschulhof** [comment]
@benjamingr thanks!

**Comment by gabrielschulhof** [comment]
Whoops! Forgot to add to `index.md`.

File: locale/en/docs/guides/abi-stability.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
+# ABI Stability
```
Comment (lpinca) [inline locale/en/docs/guides/abi-stability.md:6]: We need a header like

```
---
title: ABI Stability
layout: docs.hbs
---
```

File: locale/en/docs/guides/abi-stability.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+N-APIs are added under an experimental flag to give the community an opportunity
+to vet them in a production environment. Experimental status means that,
+although care has been taken to ensure that the new API will not have to be
+modified in an ABI-incompatible way in the future, it has not yet been
+sufficiently proven in production to be correct and useful as designed and, as
+such, may undergo ABI-incompatible changes before it is finally incorporated
+into a forthcoming version of N-API. That is, an experimental N-API is not yet
+covered by the forward compatibility guarantee.
+
+[`node_api.h`]: ../../src/node_api.h
```
Comment (lpinca) [inline locale/en/docs/guides/abi-stability.md]: I guess these links don't work.

**Comment by gabrielschulhof** [comment]
@lpinca I have addressed your comments.

**Review by lpinca — APPROVED** [review]

**Review by ghost — APPROVED** [review]

File: locale/en/docs/guides/index.md [hunk 3]
```diff
@@ -29,3 +29,4 @@ layout: docs.hbs
 - [Backpressuring in Streams](backpressuring-in-streams/)
 - [Domain Module Postmortem](domain-postmortem/)
 - [How to publish N-API package](publishing-napi-modules/)
+- [ABI Stability](abi-stability/)
```
Comment (Trott) [inline locale/en/docs/guides/index.md:32]: Is this really a module-related guide? If not, does it maybe belong in the General section instead of here?

Comment (gabrielschulhof) [reply locale/en/docs/guides/index.md:32]: Well, although it starts with generic ABI-related information, it mostly pertains to N-API. That's why I put it there.

Comment (Trott) [reply locale/en/docs/guides/index.md:32]: Works for me.

**Comment by gabrielschulhof** [comment]
Landed in 3214a8dae0fa52a62db2e81ab3aed1e485032779.

**Pull request closed without merging**